src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -434,5 +435,18 @@ internal static void SetCurrentUICulture(CultureInfo cultureInfo)
                 Thread.CurrentUICulture.SetValue(thread, cultureInfo);
             }
         }
+
+        internal static class Assembly
+        {
+            private const string TypeName = "System.Reflection.Assembly";
+
+            internal static readonly Type Type = ReflectionUtil.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            internal static readonly MethodInfo GetTypeThrowOnErrorIgnoreCase = Type
```

Comment:
The problem is that it's not clear from the call site what the booleans mean. Can't use named parameters unless I define a custom delegate type, which feels like an overkill.